docs/migrate-matomo-4-to-5.md

```diff
@@ -56,7 +56,7 @@ This will build all vue components for your plugin and must be run before testin
 
 With Matomo 5 we have introduced vendor proxy patterns. Plugins should no longer directly use any dependencies that are shipped with Matomo. We have introduced a set of classes and methods that can be used instead.
 
-This step will allow us to easily update and replace vendor libraries without the requirement to do a major release for it as plugins will be fully encapsulated and separated from our dependencies. If your plugin needs direct access to a certain dependency, consider creating a GitHub issue for adding a proxy pattern for it, or bundle your own prefixed version of the dependency with your plugin.
+This step will allow us to easily update and replace vendor libraries without the requirement to do a major release for it as plugins will be fully encapsulated and separated from our dependencies. If your plugin needs direct access to a certain dependency, consider creating a [GitHub issue](https://github.com/matomo-org/matomo/issues) for adding a proxy pattern for it, or bundle your own prefixed version of the dependency with your plugin.
```

Comment:
@sgiehl @michalkleiner @bx80 I was thinking of a couple possible improvements that we could make here to clarify things a little more for developers.
- Could we provide an example of a Composer dependency that they might want to proxy?
- What's the best method for bundling prefixed dependencies? ([PhpScoper?](https://github.com/humbug/php-scoper)) Is there a link to a blog or something we could include?

Reply:
> Could we provide an example of a Composer dependency that they might want to proxy?

I think we should have proxy classes for all relevant libraries. But maybe for any reason someone could e.g. want to use the sparkline library or the yaml parser directly.

> What's the best method for bundling prefixed dependencies? ([PhpScoper?](https://github.com/humbug/php-scoper)) Is there a link to a blog or something we could include?

Yes. That would be PhpScoper. There is a open PR in core to prefix some dependencies, which you can look at. But I did not yet have any thoughts how to best use that for a plugin only.